### PR TITLE
Status page updates

### PR DESCRIPTION
**Pull request by jknapp25**
This includes the status page updates documented [here](https://www.figma.com/file/BkeblOMtaBc6BM5TDZ0VPJ/Patient-App?type=design&node-id=760-318&mode=design&t=yDgswGSO7WCY8iNv-0) as well as some cleanup that made that easier to do. Also some other quality improvements.

- copy updates (buncha new stuff, pluralize ui, dynamic header/subheader)
- tag copy: 24hr -> Open 24 Hours
- fix: it was defaulting to pickup states if patient selected (not going directly to status view)
- show fake rx header on boson and neutron
- hide courier/mail order pharmacy details on status page
- fix: canceled and no match ui weren't surfacing
- debounce location search

### CODE REVIEW

File: apps/patient/src/components/LocationModal.tsx [hunk 1]
```diff
@@ -80,16 +89,14 @@ export const LocationModal = ({ isOpen, onClose }: any) => {
   };
 
   const loadOptions = (inputValue: string, callback: (options: any) => void) => {
```
Comment (captDaylight) [inline apps/patient/src/components/LocationModal.tsx:91]: This `loadOptions` function is just passing the variables to `debouncedSearchForLocations`, maybe get rid of it

File: apps/patient/src/views/Pharmacy.tsx [hunk 2]
```diff
@@ -340,17 +339,31 @@ export const Pharmacy = () => {
           setTimeout(() => {
             setShowFooter(false);
 
+            // Fudge it so that we can show the pharmacy card on initial load of the
+            // status view for all types. On my christmas list for 2024 is better
```
Comment (captDaylight) [inline apps/patient/src/views/Pharmacy.tsx:343]: 🎅

File: apps/patient/src/views/Pharmacy.tsx [hunk 3]
```diff
@@ -446,20 +459,23 @@ export const Pharmacy = () => {
     }
   }, [enableOpenNow, enable24Hr]);
 
+  const flattenedFills = countFillsAndRemoveDuplicates(order.fills);
```
Comment (captDaylight) [inline apps/patient/src/views/Pharmacy.tsx:462]: no need to do this in this PR, but maybe these could just be derived in the provider and passed with the useOrderContext. Seen this logic twice now

File: apps/patient/src/views/Status.tsx [hunk 4]
```diff
@@ -180,16 +165,24 @@ export const Status = () => {
     }
   }, []);
 
-  // Only show "Text us now" prompt if pickup and RECEIVED or READY
-  const showChatAlertStates: types.FulfillmentState[] = ['RECEIVED', 'READY'];
-  const showChatAlert = showChatAlertStates.includes(fulfillment?.state);
+  const flattenedFills = countFillsAndRemoveDuplicates(fills);
```
Comment (captDaylight) [inline apps/patient/src/views/Status.tsx:168]: :)

File: apps/patient/src/components/LocationModal.tsx (hunk 1 above)
Comment (jknapp25) [reply apps/patient/src/components/LocationModal.tsx:91]: cleans up the jsx a little, but I guess it's a bit superfluous, I'll remove

File: apps/patient/src/views/Pharmacy.tsx (hunk 3 above)
Comment (jknapp25) [reply apps/patient/src/views/Pharmacy.tsx]: added to context

**Pull request merged**